groovy/postInit/chemistry/inorganic_chemistry/elements/s_block/group2/BerylliumChain.groovy

```diff
@@ -308,6 +300,121 @@ DISTILLERY.recipeBuilder()
         .duration(100)
         .buildAndRegister()
 
-//BERTRANDITE CHAIN (SPECIALIZED)
+ROASTER.recipeBuilder()
+        .inputs(ore('dustBeryllium'))
+        .fluidInputs(fluid('oxygen') * 1000)
+        .outputs(metaitem('dustBerylliumOxide') * 2)
+        .EUt(Globals.voltAmps[2])
+        .duration(100)
+        .buildAndRegister()
 
+//BERTRANDITE CHAIN (SPECIALIZED)
 //HIGH PURITY CHAIN
+BR.recipeBuilder()
+        .inputs(ore('dustBertrandite') * 1)
+        .fluidInputs(fluid('sulfuric_acid') * 4100) //100 for Ammonium Alum
+        .outputs(metaitem('dustSiliconDioxide') * 6)
+        .fluidOutputs(fluid('impure_bertrandite_leach') * 1000) //Be4Si2O7(OH)2 + 4H2SO4 -> 4BeSO4 + 2SiO2 + 5H2O
+        .EUt(Globals.voltAmps[2])
+        .duration(100)
+        .buildAndRegister()
+
+BR.recipeBuilder()
+        .inputs(ore('dustAmmoniumSulfate') * 3) // 1/5 mol
+        .fluidInputs(fluid('impure_bertrandite_leach') * 2000)
+        .chancedOutput(metaitem('dustAmmoniumAlum') *4, 8000, 0) //On average it should be 3.2 because 1/5 mol
+        .fluidOutputs(fluid('bertrandite_leach') * 2000)
+        .EUt(Globals.voltAmps[2])
+        .duration(100)
+        .buildAndRegister()
+
+BLENDER.recipeBuilder()
+        .fluidInputs(fluid('ammonia_solution') * 2000) 
+        .fluidInputs(fluid('di_two_ethylhexyl_phosphoric_acid') * 2500) //DEHPA or D2EHPA
+        .fluidInputs(fluid('kerosene') * 5500)
+        .fluidOutputs(fluid('beryllium_extraction_mixture') * 10000)
+        .EUt(Globals.voltAmps[5]) //tier gating part for this chain
+        .duration(100)
+        .buildAndRegister() 
+
+CENTRIFUGE.recipeBuilder()
+        .fluidInputs(fluid('bertrandite_leach') * 1000)
+        .fluidInputs(fluid('beryllium_extraction_mixture') * 1000)
```

Comment:
the correct ratio to load an extraction mixture is 2 D2EHPA per 1 Be2+. Your formulation would require 8000 mixture per beryllium.